pageserver/src/tenant.rs

```diff
@@ -1304,41 +1304,55 @@ impl Tenant {
     }
 
     async fn load_timeline_metadata(
+        self: Arc<Tenant>,
+        timeline_id: TimelineId,
+        remote_storage: GenericRemoteStorage,
+        cancel: CancellationToken,
+    ) -> anyhow::Result<TimelinePreload> {
+        let client = RemoteTimelineClient::new(
+            remote_storage.clone(),
+            self.deletion_queue_client.clone(),
+            self.conf,
+            self.tenant_shard_id,
+            timeline_id,
+            self.generation,
+        );
+        let timeline_preload = async move {
+            debug!("starting index part download");
+
+            let index_part = client.download_index_file(&cancel).await;
+
+            debug!("finished index part download");
+
+            Result::<_, anyhow::Error>::Ok(TimelinePreload {
+                client,
+                timeline_id,
+                index_part,
+            })
+        }
+        .map(move |res| {
+            res.with_context(|| format!("download index part for timeline {timeline_id}"))
+        })
+        .instrument(info_span!("download_index_part", %timeline_id))
```

Comment:
Shouldn't this span be set in the caller...?

Reply:
If you do it here, we are now inside the task which has been spawned as opposed to being in the parent task as was before this PR. There will be no connection to the parent span from here.

Reply:
If you really insist on creating span in the method, you'll need to pass in the parent span.

Reply:
I think the more natural way would be to assert that both tenant_id and timeline_id are set whenever creating the future and then just move the span creation to caller.

Reply:
ok moved it to the caller, shrug

Reply:
Hmm it's not yet moved?